frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.unit.spec.js

```diff
@@ -1,325 +1,1651 @@
-      [value, date, printComparison(comparison)].join("; ");
-
-    const cols = [
-      DateTimeColumn({ name: "Month" }),
-      NumberColumn({ name: "Count" }),
-      NumberColumn({ name: "Sum" }),
-    ];
-    const series = ({ rows }) => [{ data: { rows, cols } }];
-    const settings = { "scalar.field": "Count" };
-
-    const testCases = [
-      {
-        description: "should correctly display percent increase",
-        rows: [
-          ["2019-10-01", 100],
-          ["2019-11-01", 300],
-        ],
-        dateUnit: "month",
-        expected: "300; Nov 2019; ↑ 200% vs. previous month: 100",
-      },
-      {
-        description: "should correctly display percent decrease",
-        rows: [
-          ["2019-10-01", 300],
-          ["2019-11-01", 100],
-        ],
-        dateUnit: "month",
-        expected: "100; Nov 2019; ↓ 66.67% vs. previous month: 300",
-      },
-      {
-        description: "should correctly display no change",
-        rows: [
-          ["2019-10-01", 100],
-          ["2019-11-01", 100],
-        ],
-        dateUnit: "month",
-        expected: "100; Nov 2019; No change vs. previous month",
-      },
-      {
-        description: "should correctly display infinite increase",
-        rows: [
-          ["2019-10-01", 0],
-          ["2019-11-01", 300],
-        ],
-        dateUnit: "month",
-        expected: "300; Nov 2019; ↑ ∞% vs. previous month: 0",
-      },
-      {
-        description:
-          "should correctly display missing data if previous value is null",
-        rows: [
-          ["2019-09-01", null],
-          ["2019-11-01", 300],
-        ],
-        dateUnit: "month",
-        expected: "300; Nov 2019; N/A vs. previous month: (empty)",
-      },
-      {
-        description: "should correctly display missing data for single row",
-        rows: [["2019-11-01", 300]],
-        dateUnit: "month",
-        expected: "300; Nov 2019; N/A vs. previous month: (empty)",
-      },
-      {
-        description:
-          "should correctly display percent decrease over missing row",
-        rows: [
-          ["2019-09-01", 300],
-          ["2019-11-01", 100],
-        ],
-        dateUnit: "month",
-        expected: "100; Nov 2019; ↓ 66.67% vs. Sep: 300",
-      },
-      {
-        description: "should correctly display percent decrease over null data",
-        rows: [
-          ["2019-09-01", 300],
-          ["2019-10-01", null],
-          ["2019-11-01", 100],
-        ],
-        dateUnit: "month",
-        expected: "100; Nov 2019; ↓ 66.67% vs. Sep: 300",
-      },
-      {
-        description:
-          "should correctly fallback to day unit if backend doesn’t return valid unit",
-        rows: [
-          ["2018-09-01", 100],
-          ["2019-11-01", 300],
-        ],
-        dateUnit: null,
-        expected: "300; Nov 1, 2019; ↑ 200% vs. Sep 1, 2018: 100",
-      },
-    ];
+    const series = ({ rows, cols }) => [{ data: { rows, cols } }];
```

Comment:
Can we use `createMockSeries` and other test helpers? It will make it easy converting this to typescript